### ISSUE TITLE
SocialPlugin fails to use environment variables

### ISSUE THREAD

**Issue by jasonren0403** [squidfunk/mkdocs-material#4819]
### Bug description

Social plugin is not working when using environment variables in plugins.social.cards

### Related links

- [Bug reporting](https://squidfunk.github.io/mkdocs-material/bug-report/)
- https://squidfunk.github.io/mkdocs-material/setup/setting-up-social-cards/#built-in-social-plugin said that you should set environment variable if you want to switch plugin off for local builds

### Reproduction

[example.zip](https://github.com/squidfunk/mkdocs-material/files/10335643/example.zip)

### Steps to reproduce

set `plugins.social.cards` to `!ENV [CARDS, false]`, just like what document says, then an Exception is encountered:
```
AttributeError: 'SocialPlugin' object has no attribute '_image_promises'
```

An minimum `mkdocs.yml` should be this:
```yml
site_name: Test site

theme:
  name: material

plugins:
  - social:
      cards: !ENV [CARDS, false]
  - search
```

### Browser

Chrome, Safari, Edge, Firefox, Opera

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/bug-report/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) or [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/bug-report/#remove-customizations) before submitting this bug report.
- [X] I have attached a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/bug-report/reproduction/).

**Comment by squidfunk** [comment]
Thanks for reporting. It has nothing to do with environment variables but fails when `cards: false` is set. This is definitely related to the latest performance optimizations, should be an easy fix.

**Comment by squidfunk** [comment]
Fixed in c8fb426d9.

**Comment by squidfunk** [comment]
Released as part of [9.0.2](https://github.com/squidfunk/mkdocs-material/releases/tag/9.0.2).
